include/sisl/auth_manager/token_client.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <string>
+
+namespace sisl {
+
+// Interface to get a token for authorization
+
+class TokenClient {
+public:
+    virtual ~TokenClient() = default;
+
+    virtual std::string get_token() = 0;
```

Comment:
`const`

Reply:
This is only ever implemented by Mock for unit tests. The `real` implementation will be in the new library I created, which is ebay specific.
https://github.corp.ebay.com/SDS/auth_manager/tree/inherit_from_sisl/src/include/auth_manager
The idea is if the user wants a token based auth system, they implement these interfaces and pass it along to grpc client and server. Any implementation would be ebay specific and cannot live here in sisl